podcasts/SleepTimerManager.swift

```diff
@@ -89,13 +89,38 @@ class SleepTimerManager {
         NotificationCenter.default.addObserver(self, selector: #selector(playbackTrackChanged), name: Constants.Notifications.playbackTrackChanged, object: nil)
     }
 
+    private var playbackCheckTimer: Timer?
+    private var recordedPlayedUpTo: Double?
+
+    // Here we watch the current episode until it actually starts playing to
+    // restart the timer per episode.
+    // This avoid issues because the Sleep Timer per episode pauses on the end
+    // of the last episode.
```

Comment:
I'm not sure this comment conveys the complexity of restarting the sleep timer. I tried many different events and logic, and ultimately everything — besides this solution — failed.

I'll be more verbose here to try to explain:

1. When we finish a sleep timer per episode, we play at the end of the last played episode;
2. When we tap play again, this episode will still be queued to play and the timer will restart
3. However, we can't differentiate if this episode "playing" is what was playing before or not. And due to the nature of `PlaybackManager`, `currentEpisode()` already changes to the next one (even though the whole player displays the previous one).
4. If we restart the timer, this "end of episode" will be counted as a one-played episode, which leads to: if the timer is set to 1 episode, nothing happens. The app thinks that this episode that just finished was the episode meant to be played. If it's bigger than 1 episode, 1 will be deduced and the timer will start with one less episode.
5. As none of this is enough, buffering is also part of the equation: we can't restart the timer unless an episode is playing (thus the check here).

I spent a good amount of hours trying to make that simpler but I couldn't. So any suggestion on improving this comment — if it's not clear — it's appreaciated.